internal/app/loki-multi-tenant-proxy/server.go

```diff
@@ -9,25 +9,49 @@ import (
 
 	"github.com/giantswarm/loki-multi-tenant-proxy/internal/pkg"
 	"github.com/urfave/cli/v2"
+	"go.uber.org/zap"
 )
 
 // Serve serves
 func Serve(c *cli.Context) error {
 	lokiServerURL, _ := url.Parse(c.String("loki-server"))
-	serveAt := fmt.Sprintf(":%d", c.Int("port"))
+	addr := fmt.Sprintf(":%d", c.Int("port"))
 	authConfigLocation := c.String("auth-config")
 	authConfig, _ := pkg.ParseConfig(&authConfigLocation)
 	authConfig.KeepOrgID = c.Bool("keep-orgid")
 
-	http.HandleFunc("/", createHandler(lokiServerURL, authConfig))
-	if err := http.ListenAndServe(serveAt, nil); err != nil {
-		log.Fatalf("Loki multi tenant proxy can not start %v", err)
+	logLevel := c.String("log-level")
+	if logLevel == "" {
+		logLevel = "INFO"
+	}
+
+	zapConfig := zap.NewProductionConfig()
+	level, err := zap.ParseAtomicLevel(logLevel)
+	if err != nil {
+		panic(fmt.Sprintf("Could not parse log level %v", zap.Error(err)))
+	}
+	zapConfig.Level = level
+
+	logger := zap.Must(zapConfig.Build())
```

Comment:
Is there a way to handle error here ?

Reply:
Well, i'd rather we panic if flag is incorrect. How would you handle it ?

Reply:
I would rather have something like
```suggestion
	logger, err := zapConfig.Build()
	if err != nil {
	    log.Print("Could not construct logger", zap.Error(err))
	    return err
	}
```